### PR TITLE
win_wait_for: added module

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Added win_wait_for module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
win_wait_for

##### ANSIBLE VERSION
```
ansible 2.4.0 (win_wait_for-module ce3cb28426) last updated 2017/07/08 15:20:54 (GMT +1000)
  config file = None
  configured module search path = [u'/home/jborean/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jborean/dev/ansible/lib/ansible
  executable location = /home/jborean/dev/ansible/bin/ansible
  python version = 2.7.13 (default, Jun 28 2017, 21:13:49) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

### CODE REVIEW

**Comment by dagwieers** [comment]
+label new_module

**Comment by dagwieers** [comment]
So the module freeze deadline for v2.4 is getting closer (2017-08-29). Now is the time to finish up, get it reviewed and merged with no delay.

**Comment by mattclay** [comment]
CI failure in integration tests:

```
{
"attempts": 300, 
"changed": false, 
"elapsed": 300.31624619999997, 
"failed": true, 
"msg": "timeout while waiting for file C:\\ansible\\win_wait_for\\test.txt to be absent"
}
```

**Comment by jhawkesworth** [comment]
since the ci failure was a timeout just going to close and open to re-trigger in case it was transient failure.

**Comment by jborean93** [comment]
@jhawkesworth, there is an underlying issue with the test, sometimes an async task fails on a host causing a failure altogether. I'm currently trying to figure out what is happening as it only occurs sporadically with AWS hosts and not locally.

**Comment by dagwieers** [comment]
I was wondering the other day if it could be useful to give **wait_for** functionality to wait for specific events to happen in Windows. This seems to be a much more convenient way in the Windows world.

Or maybe this ought to go into a more specialized **win_wait_for_event** module given the plethora of options related to this ?

**Comment by jborean93** [comment]
That would be useful to have, I'm probably leaning towards separating them into separate modules, this was complex enough to decipher the various options that were required for file and port checks. I didn't split it up at the time as it is really just to have a closer parity to `wait_for`.

**Comment by dagwieers** [comment]
Yes, don't get me started on that. All modules that try to allow doing more than one thing with parameters, end up being a nightmare, especially if these parameters have a different meaning/effect when being used. One of them is the timeout option, which normally is the error-condition when reached, except if it is the only parameter provided.

Flawed design in my opinion. We do it to ourself, we do, and that's what really hurts...

File: lib/ansible/modules/windows/win_wait_for.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_wait_for
+version_added: '2.4'
+short_description: waits for a condition before continuing
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_wait_for.py]: short_description usually starts with uppercase. But I assume this was copied from the original.

File: lib/ansible/modules/windows/win_wait_for.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+description:
+- You can wait for a set amount of time C(timeout), this is the default if
+  nothing is specified.
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_wait_for.py]: The original documentation was updated recently wrt. using only timeout. You may want to update the wording here too.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_wait_for.py (hunk 1 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_wait_for.py]: It was not, my mistake and will fix up.

File: lib/ansible/modules/windows/win_wait_for.py (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_wait_for.py]: Will have a look thanks.

Edit: Still looks the same as this sentence.

**Comment by jborean93** [comment]
I am still going to merge this before the freeze even if the tests are temporarily fail due to async issue. The tests are still there but don't run in Shippable until this is ultimately fixed.

**Pull request merged**